### PR TITLE
Remove gravityID in favor of internalID and slug

### PR DESCRIPTION
**Pull request by zephraph**
Blocked by https://github.com/artsy/metaphysics/pull/1785.

This insanely huge PR is the result of work being done in https://github.com/artsy/metaphysics/pull/1785. In tackling metaphysics V2 we're both removing `__id` so that we can update relay _and_ trying to remove ambiguities in the api. 

At first we created a `gravityID` that was used for fields from gravity that could either be an id or a slug. Thing is, a field will _never_ be both. It's an either or scenario. So, in the linked metaphysics PR we've removed the notion of `gravityID` for a simplified model. 

Now there are three id-ish fields

1. `id` - A globally unique identifier for a type
2. `internalID` - An id that likely maps to a database id in a given service
3. `slug` - A human readable id often used for urls

If you're reviewing this, use the commits as a guide. The first commit is the relay types update, the second are the transformations that I'm confident about, the third are transformations I'm not at all confident about. 

#skip_new_tests

### CODE REVIEW

File: src/lib/tests/__tests__/MockRelayRendererFixtures.tsx [hunk 1]
```diff
@@ -26,7 +26,8 @@ export const Artwork = createFragmentContainer(
     <View>
       <Image source={{ uri: props.artwork.image.url }} />
       <Metadata artworkMetadata={props.artwork} />
-      {props.artwork.artist && <ArtistQueryRenderer id={props.artwork.artist.gravityID} />}
+      {/* FIXME: Should this be a slug? */}
+      {props.artwork.artist && <ArtistQueryRenderer id={props.artwork.artist.slug} />}
```
Comment (ashfurrow) [inline src/lib/tests/__tests__/MockRelayRendererFixtures.tsx]: This ultimately is used with the `artist(id: ...)` MP resolver, so whichever works there 👍

**Comment by zephraph** [comment]
Ideally I'd like to verify all the FIXME cases and remove those entries after either we decide if it's correct or needs to be updated.

File: src/lib/Scenes/Fair/Screens/FairArtworks.tsx [hunk 2]
```diff
@@ -23,12 +23,13 @@ interface State {
...
   context_screen: Schema.PageNames.FairAllArtworksPage,
   context_screen_owner_type: Schema.OwnerEntityTypes.Fair,
-  context_screen_owner_slug: props.fair.gravityID,
+  context_screen_owner_slug: props.fair.slug,
   context_screen_owner_id: props.fair.internalID,
 }))
 export class FairArtworks extends React.Component<Props, State> {
   handleRefetch = params => {
-    const { gravityID: fairID } = this.props.fair
+    // FIXME: Should this be id? It might have been internalID previously
```
[the remark this answers is not in the document]
Comment (zephraph) [reply src/lib/Scenes/Fair/Screens/FairArtworks.tsx]: Looking at metaphysics it looks like it can be an id or slug. I'm just gonna use internalID.

**Comment by zephraph** [comment]
@ashfurrow I'd be down to pair on it next week sometime. I went ahead and made the updates you suggested. 

We'll definitely want to do a lot of testing, for sure.

**Comment by ashfurrow** [comment]
@zephraph sounds good 👍

**Comment by zephraph** [comment]
@alloy there's a test failure here saying that `PartnerShowEventType` is unknown. It's referencing back to `PartnerShowEvent` which for sure doesn't exist anymore. 

I was thinking that perhaps it's a cache thing, but it's happening locally _and_ on CI. I'm not sure exactly how it keeps finding that type, but it's nowhere left over in the emission codebase that I can find. 

Thoughts?

**Comment by alloy** [comment]
Let's pair on that later today!

**Pull request merged**